path.go

```diff
@@ -159,6 +162,11 @@ func (p Path) HasPrefix(other Path) bool {
 	return strings.HasPrefix(p.value, other.value)
 }
 
+// Equal checks if the path is the same as the other path.
+func (p Path) Equal(other Path) bool {
```

Comment:
I think that the best solution would be to change `apiequality.Semantic.DeepEqual` so that it understands `T Equal(T)` method just as `github.com/google/go-cmp/cmp` does.